src/lib/connect/index.ts

```diff
@@ -45,33 +49,35 @@ function parseAuthOptions(opts) {
 
 /**
  * connect - connect to an MQTT broker.
- *
- * @param {String} [brokerUrl] - url of the broker, optional
- * @param {Object} opts - see MqttClient#constructor
  */
-function connect(brokerUrl, opts) {
+export default function connect(
+	brokerUrl: string | IClientOptions,
+	opts?: IClientOptions,
+): MqttClient {
```

Comment:
maybe do function overloading here? one signature with just an opts param and another signature with the url string and opts

Reply:
@vishnureddy17 I have been able to add them, check: [`1c56921` (#1641)](https://github.com/mqttjs/MQTT.js/pull/1641/commits/1c569211bf6aaa961e13b89f877d4677ae64278a)